File: ws/chavez/u2/ws17UnitTest/calculator/src/test/java/ec/edu/espe/calculator/controller/BasicOperationTest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
 */
package ec.edu.espe.calculator.controller;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Odalys Chavez, CodeBreakers, @ESPE
 */
public class BasicOperationTest {
    
    public BasicOperationTest() {
    }

    @org.junit.jupiter.api.Test
    public void testSomeMethod() {
        // TODO review the generated test code and remove the default call to fail.
        fail("The test case is a prototype.");
    }

    /**
     * Test of add method, of class BasicOperation.
     */
    @Test
    public void testAdd() {
        System.out.println("add");
        float addend1 = 1.2F;
        float addend2 = 2.4F;
        float expResult = 3.6F;
        float result = BasicOperation.add(addend1, addend2);
        assertEquals(expResult, result, 0);
        // TODO review the generated test code and remove the default call to fail.
        //fail("The test case is a prototype.");
    }

    /**
     * Test of subtract method, of class BasicOperation.
     */
    @Test
    public void testSubtract() {
        System.out.println("subtract");
        float minuend = 3.6F;
        float subtrahend = 1.2F;
        float expResult = 2.4F;
        float result = BasicOperation.subtract(minuend, subtrahend);
        assertEquals(expResult, result, 0);
        // TODO review the generated test code and remove the default call to fail.
        //fail("The test case is a prototype.");
    }
    
}

File: hw/basurto/u2/hw11UnitTest/JavaScript/Calculator/test/basicOperation.test.js
const BasicOperation = require('../controller/basicOperation');

describe('BasicOperation Tests', () => {

    test('test add', () => {

        console.log("add");

        let addend1 = 2.4;
        let addend2 = 1.2;

        let expectedResult = 3.6;

        let result = BasicOperation.add(addend1, addend2);

        expect(result).toBeCloseTo(expectedResult);

    });

    test('test subtract', () => {

        console.log("subtract");

        let minuend = 3.6;
        let subtrahend = 1.2;

        let expectedResult = 2.4;

        let result = BasicOperation.subtract(minuend, subtrahend);

        expect(result).toBeCloseTo(expectedResult);

    });

});